hypha/apply/funds/tables.py

```diff
@@ -182,30 +183,46 @@ def render_role_icon(self, value):
         return ''
 
 
-def get_used_rounds(request):
+def get_used_rounds(request=None, queryset=None):
+    if queryset:
```

Comment:
The queryset seems to always have a value.

Also on larger datasets all the "if queryset" versions crash the server with a "Proxy Error".

Reply:
>The queryset seems to always have a value.

Actually, it is a bit tricky part, we are calling it from `__init__` and [updating the filter's field queryset](https://github.com/HyphaApp/hypha/pull/3003/files#diff-d8cf87355e041ebbd15a4725ce9b462fc6c1e3d752f8c1fee9aeecbd6a56cb98R303) to a value instead of a  callable that's why it is not getting called internally from forms and it always has a queryset. 
In the near future, if somebody removes any of the filter queryset update statement, it will get called internally and execute without queryset. That's why I put a check for queryset but if it is taking time, I think we can use separate methods for [initialization](https://github.com/HyphaApp/hypha/pull/3003/files#diff-d8cf87355e041ebbd15a4725ce9b462fc6c1e3d752f8c1fee9aeecbd6a56cb98R274) and to update the filter's field queryset on the basis of the request's queryset.

Reply:
Also, queryset here has 2 reference, so it is a bit confusing. I'll change request's or view's 'queryset' to 'dataset' just for inside the table.